### DISCUSSION TITLE
Support for git switch/restore?

### DISCUSSION THREAD

**Discussion by iliakan** [libgit2/libgit2#6484] [Q&A]
Are there any plans to support git switch and restore commands?

They are marked experimental, but in fact very stable in Git since 2019.

P.S. How compatible is libgit2 with git overall? 
Is there any information about missing/different options?

**Comment by ethomson** [comment]
> Are there any plans to support git switch and restore commands?

These are high-level commands that are facades over existing functionality in `git checkout`. In libgit2, you can switch and restore using `git_checkout` today.

It's not impossible that we would provide another API to do the same thing, but it's not impossible.

> P.S. How compatible is libgit2 with git overall?

There are features that we do not support -- for example, reachability bitmaps or shallow clone -- but the features that we do support we aim to be bug-for-bug compatible.
